IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
 
 class BaseService<T: PersistenceObject>  {
     
-    private let persistence: Persistence
-    private let restClient: AnyObject? = nil
-    
-    //TODO: we should inclued the API access also
-    init(persistence: Persistence) {
+    private let persistence: Persistence!
+    internal let api: Api!
+
+    init(persistence: Persistence, api: Api) {
         self.persistence = persistence
+        self.api = api
     }
     
     public func getAll() -> [T] {
         return persistence.getAll(entityName: T.EntityName)
     }
     
-    public func getEntityBy(identifier: Int) -> T {
+    public func getEntityBy(identifier: String) -> T? {
         return persistence.getBy(entityIdentifier: identifier, entityName: T.EntityName)
     }
     
+    public func createEntity(json: JSON) {
```

Comment:
This is a bit weird..

So, we have

App
'--------'
Domain
'--------'
Infra

`json` belongs to infra, but this class is exposing a method to be used by `App` layer (because public methods are consumed by the upper layer, otherwise they should be protected/private/internal-C#-style). Anyone consuming this method outside a service will be breaking the layered architecture.

On the other hand, it seems you are using it only within Domain layer, and more precisely, inside a child class,, so this method should be protected, and that way we prevent weird scenarios were a ViewModel ends up handling `json`.